src/ORM/Connect/MySQLiConnector.php

```diff
@@ -382,4 +392,38 @@ public function getLastError()
         }
         return $this->dbConn->error;
     }
+
+    public function getLastErrorCode(): int
+    {
+        // Check if a statement was used for the most recent query
+        if ($this->lastStatement && $this->lastStatement->errno) {
+            return $this->lastStatement->errno;
+        }
+        return $this->dbConn->errno;
+    }
+
+    /**
+     * Throw the correct DatabaseException for this error
+     *
+     * @throws DatabaseException
+     */
+    private function throwRelevantError(string $message, int $code, int $errorLevel, ?string $sql, array $parameters): void
```

Comment:
Change to throwRelevantException.  databaseError() is poorly named as it throws an exception.  So does duplicateEntryError(), though I've recommended we renamed that in this PR

Reply:
`databaseError()` can use `E_USER_ERROR`. `throwRelevantException()` calls `databaseError()` and therefore won't always throw an exception. I'll leave this as is.